src/NServiceBus.RavenDB/Outbox/OutboxPersister.cs

```diff
@@ -78,12 +78,14 @@ public Task Store(OutboxMessage message, OutboxTransaction transaction, ContextB
                 index++;
             }
 
-            return session.StoreAsync(new OutboxRecord
+            var outboxRecord = new OutboxRecord
             {
                 MessageId = message.MessageId,
                 Dispatched = false,
                 TransportOperations = operations
-            }, GetOutboxRecordId(message.MessageId));
+            };
+            await session.StoreAsync(outboxRecord, GetOutboxRecordId(message.MessageId)).ConfigureAwait(false);
```

Comment:
According to https://ravendb.net/docs/article-page/4.2/csharp/client-api/session/how-to/get-and-modify-entity-metadata the entity needs to be tracked first before it is possible to access the metadata dictionary if I understood correctly. Therefore I switched to use async/await